test/AutoExpandbleMap/AutoExpandble_map_utest.cpp

```diff
@@ -2,81 +2,81 @@
 
 #include "Map_test.h"
 
-TEST(AutoExpandbleMapBaseTests, Right) {
-  HeWi param = AutoExpandbleMap_Test({2, 0});
-  bool flag = (param.height == 1)&&(param.width == 4);
-  EXPECT_TRUE(flag);
-  flag = (param.x == 0)&&(param.y == 0);
-  EXPECT_TRUE(flag);
+TEST(AutoExpandbleMapBaseTests, Expand_Right) {
+  MapParam param = test_auto_expand({2, 0});
```

Comment:
Have you tried to write a macro/function for the expansion verification code?
E.g. test_expansion(point, expected map params).
Have you also considered implementation of operator== for MapParams?